airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/MetricsRegistries.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.lib;
+
+import com.google.api.client.util.Preconditions;
+
+/**
+ * Enum source of truth of all Airbyte metrics. Each enum value represent a metric and is linked to
+ * an application and contains a description to make it easier to understand.
+ *
+ * Each object of the enum actually represent a metric, so the Registry name is misleading. The
+ * reason 'Registry' is in the name is to emphasize this enum's purpose as a source of truth for all
+ * metrics. This also helps code readability i.e. AirbyteMetricsRegistry.metricA.
+ *
+ * Metric Name Convention (adapted from
+ * https://docs.datadoghq.com/developers/guide/what-best-practices-are-recommended-for-naming-metrics-and-tags/):
+ * <p>
+ * - Use lowercase. Metric names are case sensitive.
+ * <p>
+ * - Use underscore to delimit names with multiple words.
+ * <p>
+ * - No spaces. This makes the metric confusing to read.
+ * <p>
+ * - Avoid numbers. This makes the metric confusing to read. Numbers should only be used as a
+ * <p>
+ * - Add units at name end if applicable. This is especially relevant for time units.
+ * <p>
+ * - Include the time period in the name if the metric is meant to be run at a certain interval.
+ */
+public enum MetricsRegistries implements MetricsRegistry {
```

Comment:
nit: Instead of `MetricsRegistries` as the enum name, what about `MetricsRegistryEnum` or `MetricsRegistryImpl`? The plural "registries" feels slightly off to me

Reply:
What about `PublicMetricRegistry` or `OssMetricRegistry`? The practice of adding the java type to the name is a throwback to when IDEs didn't provide introspection. I generally feel its superfluous and make it more difficult to read.

Reply:
Sounds good - I was following MetricEmittingApp & MetricEmittingApps pattern. I think the Oss one makes sense.

Reply:
makes sense. i suppose that pattern is less logical here since it suggests we have multiple registries. appreciate trying to keep the consistency!